### PR TITLE
feat(profile): synchronize delete-profile with CRM

### PR DESCRIPTION
**Pull request by anette1983**
## Describe your changes

- Added synchronization of delete profile functionality with the CRM system. Now, when a user deletes his profile, the changes are reflected not only in our database but also in the CRM.

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://testhomework4.atlassian.net/jira/software/projects/AG/boards/6?selectedIssue=AG-182)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [ ] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Review by AKutusenko — APPROVED** [review]

File: src/sections/user/user-del-dialog.tsx [hunk 1]
```diff
@@ -45,9 +50,16 @@ export default function DeleteProfileDialog({ id }: Props) {
   const handleDeleteUser = async (userId: string) => {
     try {
       await deleteUser({ id: userId });
+      await deleteAgent({ id: qobrixAgentId }).unwrap();
+      await deleteContact({ id: qobrixContactId }).unwrap();
+
       localStorage.removeItem(StorageKey.TOKEN);
+
       enqueueSnackbar('User deleted succesfully!', { variant: 'success' });
```
Comment (lyudad) [inline src/sections/user/user-del-dialog.tsx]: https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798 1.3.3

File: src/sections/user/user-del-dialog.tsx [hunk 2]
```diff
@@ -45,9 +50,16 @@ export default function DeleteProfileDialog({ id }: Props) {
   const handleDeleteUser = async (userId: string) => {
     try {
       await deleteUser({ id: userId });
+      await deleteAgent({ id: qobrixAgentId }).unwrap();
+      await deleteContact({ id: qobrixContactId }).unwrap();
```
Comment (lyudad) [inline src/sections/user/user-del-dialog.tsx]: мне не нравится такое решение, если нужно будет еще какие то связи удалять, нужно будет фронт менять,  возможно ендпоинты добавлять. я бы сделала 1 ендпоинт на беке - delete account и там удалила бы уже все зависимости, к тоже сервисы уже есть

File: src/sections/user/user-del-dialog.tsx (hunk 1 above)
Comment (anette1983) [reply src/sections/user/user-del-dialog.tsx]: fixed

File: src/sections/user/user-del-dialog.tsx (hunk 2 above)
Comment (anette1983) [reply src/sections/user/user-del-dialog.tsx]: Поправила

**Review by lyudad — APPROVED** [review]

**Pull request merged**